sapicore/engine/component/__init__.py

```diff
@@ -58,6 +58,10 @@ def __init__(self, identifier: str = None, configuration: dict = None, device: s
         self.simulation_step = 0
         self.dt = DT
 
+        # we don't know what attributes derivative component classes might introduce, but we want them initialized.
```

Comment:
This is kinda scary because it means this can happen:

```py

class MyComponent(Component):
    the_names = "fruit"
    def print_me(self):
        print(self.the_names)

comp = MyComponent(the_name="apple")
comp.print_me()
```
Would print `fruit`. But you'd expect `apple`. Blindly setting every arg we were given can cause all kinds of heavoc that way. And there's never an API for a class because any property may or may not be part of the class, depending on how the class was initialized. And with subclasses, similarly, what was at one point an arg for the class may suddenly become just a property.

The better way is to have to pass in a dict, that was save. E.g. 

```py
def __init__(self, identifier: str = None, configuration: dict = None, device: str = "cpu", options: dict | None = None):
    self.options = options

def __getattr__(self, name):
    if name in self.options:
        return self.options[name]
    else:
        raise AttributeError

def __setattr__(self, name, value):
    if name in self.options:
        self.options[name] = value
    else:
        object.__setattr__(self, name, value)
```

Although even this is a bit scary to do for a root component that is used everywhere.

Reply:
Great suggestion. Let's merge for now to keep things consistent with the version used in the SciRep manuscript. We can open an issue and implement the fix in the next PR (or come up with a more comprehensive solution).